### ISSUE TITLE
Issue with Drawer component mounting and un-mounting child content twice

### ISSUE THREAD

**Issue by jairo-vera-21** [jpmorganchase/salt-ds#3153]
### Package name(s)

Core (@salt-ds/core)

### Package version(s)

@salt-ds/core": "^1.21.0",

### Description

When closing a Drawer component, the child component representing the content is being mounted and un-mounted a 2nd time. This is causing our API call that fires on the component mount to execute twice.

I noticed this behavior when migrating the Drawer component from the Salt/Lab library to the Salt/Core v1.21.0 library. This never occurred when we were using the Drawer from the Lab library.

I have attached sample code that reproduced the issue with console logging included.

Steps to reproduce:
1. Run the code
2. Click on button to open the Drawer
3. Click on "x" button to close Drawer
4. Observe the logs to see the additional mount/unmounts

### Steps to reproduce

```
import React, { useState, useEffect, useCallback, MouseEventHandler,} from 'react';
import {
  Button,
  StackLayout,
  H2,
  Text,
  useId,
  Drawer,
  DrawerCloseButton,
} from '@salt-ds/core';
import './score-card.scss';

const ScoreCards = () => {
  const [open, setOpen] = useState(false);
  const id = useId();

  const handleRequestOpen = () => {
    console.log('Opening the Drawer');
    setOpen(true);
  };

  const handleClose = useCallback(() => {
    console.log('Closing the Drawer');
    setOpen(false);
  }, []);

  return (
    <>
      <Button onClick={handleRequestOpen}>Open Right Drawer</Button>
      <Drawer
        open={open}
        position="right"
        style={{ width: 500 }}
        aria-labelledby={id}
      >
        <DrawerBody id={id} handleClose={handleClose} />
      </Drawer>
    </>
  );
};

const DrawerBody = ({
  id,
  handleClose,
}: {
  id: string;
  handleClose: Function;
}) => {
  useEffect(() => {
    console.log('Drawer Body Mounted');
    return () => {
      console.log('Drawer Body Un-Mounted');
    };
  }, []);

  return (
    <StackLayout>
      <DrawerCloseButton onClick={handleClose as MouseEventHandler} />
      <H2 id={id}>Section Title</H2>
      <Text>Hello World!</Text>
    </StackLayout>
  );
};

export default ScoreCards;

```

### Expected behavior

The child content should only need to mount once (when the drawer is opened) and un-mount once (when the drawer is closed).

### Operating system

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

**Comment by joshwooding** [comment]
Thanks for reporting this @jairo-vera-21 we've released @salt-ds/core@1.22.0 with a fix

**Comment by jairo-vera-21** [comment]
Thank you Salt team for the quick response and resolution. I can confirm the Drawer is no longer double mounting/unmounting after installing Salt/Core v1.22.0
